test/platform/test_linux_amd64_compatibility.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+BLUE_FG=$(tput setaf 4)
+GREEN_FG=$(tput setaf 2)
+RED_FG=$(tput setaf 1)
+TEAL_FG=$(tput setaf 6)
+END_FG_COLOR=$(tput sgr0)
+
+OS_LIST=(
+    centos:7
+    centos:8
+    fedora:28
+    ubuntu:16.04
+    ubuntu:18.04
+)
+
+FAILED=()
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+PLATFORM=$($SCRIPTPATH/../../scripts/osarchtype.sh)
+
+if [ "${PLATFORM}" != "linux/amd64" ]
+then
+    echo "The test_linux_amd64_compatibility.sh script is intended to support local execution only on linux/x86-64 machines."
+    exit 0
+fi
+
+build_images () {
+    # We'll use this simple tokenized Dockerfile.
+    # https://serverfault.com/a/72511
+    IFS='' read -r -d '' TOKENIZED <<EOF
+FROM {{OS}}
+
+WORKDIR /root
+CMD ["/bin/bash"]
+EOF
+
+    for item in ${OS_LIST[*]}
+    do
+        WITH_PACMAN=$(echo -e "${TOKENIZED}")
+
+        echo -e "$BLUE_FG[$0]$END_FG_COLOR Testing $item..."
+
+        echo -e "${WITH_PACMAN/\{\{OS\}\}/$item}" > Dockerfile
```

Comment:
How about just giving echo `$TOKENIZED` here instead of creating the temp var `WITH_PACKMAN`?